### ISSUE TITLE
Refactoring Parameter class

### ISSUE THREAD

**Issue by justusschock** [delira-dev/delira#6]
I (and @ORippler ) would like to refactor the `HyperParameters` class.

Currently we have all parameters splitted in `HyperParameters` and `model_kwargs`. 
For they hyperparameter search we must decide, whether a parameter is variable or not, and we must combine the variable `hyper_parameters` with the variable `model_kwargs` and the fixed `hyper_parameters` with the fixed `model_kwargs`. 

Our proposal is to unify them in a class `Parameter` which is dict-like (as our actual hyperParameter class) but contains a few restrictions:

We would introduce fixed sections for model- and trainer-parameters and fixed sections for fiexed and variable parameters.

The whole Structure would either look like 
```
Parameters = {
    "model": {
        "variable":{
            params
        }
        "fixed":{
            params
        }
    }
    "trainer":{
          "variable":{
            params
        }
        "fixed":{
            params
        }
    }
}   
```

or like 

```
Parameters = {
    "variable": {
        "model":{
            params
        }
        "trainer":{
            params
        }
    }
    "fixed":{
          "model":{
            params
        }
        "trainer":{
            params
        }
    }
}   
```

What do you think @haarburger  @ORippler  ?

**Comment by ORippler** [comment]
Based on [wikipedia](https://en.wikipedia.org/wiki/Hyperparameter_(machine_learning)), we could use the term `Hyperparameters` instead of `Parameters`.

I also prefer this solution due to its high readability.

**Comment by cwern1** [comment]
There are so many definitions of `Hyperparameters` that I would actually prefer `Parameters`.

**Comment by justusschock** [comment]
solved by 2ca2364d2c5cb94dc2ab1b0bffefd57b6c1f526c
